### ISSUE TITLE
How to use external matrix inside nn module ?  

### ISSUE THREAD

**Issue by antonurf** [torch/DEPRECEATED-torch7-distro#212]
Hello,
I'm facing an issue when I try to multiply two Torch tensors involved in modules : one is an external matrix and the second is the evaluation of the output of torch nn module.
Below is the code and the error I got.
There is probably an easier way to deal with external input inside neural network modules 
If someone knows how to deal with this problem, it would be really helpful.
Thanks a lot,

shuffling=torch.ones(10,10) -- external matrix
input=torch.ones(10)

factor_Y=nn.Sequential() 
factor_Y:add(nn.Linear(10,10))

model_shuffling=nn.Sequential(); 
model_var1=nn.ParallelTable() 
model_var1:add(factor_Y)
model_var1:add(nn.Identity())
model_shuffling:add(model_var1)
model_shuffling:add(nn.MM())

test=model_shuffling:forward({input,shuffling})

and the error message : 
"input tensor must be 2D or 3D " (speaking about nn.MM() )
